rs/rosetta-api/icrc1/tests/golden_state_upgrade_downgrade.rs

```diff
@@ -121,16 +138,29 @@ fn should_upgrade_icrc_sns_canisters_with_golden_state() {
         YUKU,
     ];
 
-    let state_machine = new_state_machine_with_golden_sns_state_or_panic();
+    let state_machine =
+        ic_nns_test_utils_golden_nns_state::new_state_machine_with_golden_sns_state_or_panic();
 
-    for canister_id_and_name in canister_id_and_names {
-        upgrade_canister(&state_machine, canister_id_and_name, ledger_wasm.clone());
+    for (canister_id_str, canister_name) in canister_id_and_names {
+        let canister_id =
+            CanisterId::unchecked_from_principal(PrincipalId::from_str(canister_id_str).unwrap());
+        // TODO: Uncomment once mainnet ledgers have been upgraded to include `ledger_num_approvals` metric
+        // ic_icrc1_ledger_sm_tests::in_memory_ledger::verify_ledger_state(&state_machine, canister_id, None);
+        upgrade_canister(
+            &state_machine,
+            (canister_id_str, canister_name),
+            ledger_wasm.clone(),
+        );
         // Upgrade again with bumped wasm timestamp to test pre_upgrade
         upgrade_canister(
             &state_machine,
-            canister_id_and_name,
+            (canister_id_str, canister_name),
             bump_gzip_timestamp(&ledger_wasm),
         );
+        ic_icrc1_ledger_sm_tests::in_memory_ledger::verify_ledger_state(
```

Comment:
Can we check this in the other two tests as well?

Reply:
Unfortunately this won't work out-of-the-box for the ckBTC and ckETH ledgers due to the issue at the end of last year, where the spender field was not stored for some burn operations. I have [a follow-up PR](https://github.com/dfinity/ic/pull/721) that will add the state verification for those ledgers also, with some workarounds.